Feature/nofiticatonScreen functionalities

WretchedBunny opened the pull request.

WretchedBunny committed:
chore(notifications): Deleted NotificatonDatasource, NotificationRemoteDataSouce, NotificationsRepository

Moved getSentListFlow to FriendShipDataSource because i did not see sense in keeping those classes. This function belongs to Friendship, because it is with FriendShip request and nothing else.

WretchedBunny committed:
feat: Notification Screen functionality
1. Added func. to Accept / Decline buttons in request row.
2. Added GeomateAlertDialog.kt composable for decline

WretchedBunny committed:
chore: Added two methods for Friends retrieval for curr. user.
1. In FriendshipRepository.kt added getFriendsAsFlow() & getFriends() for retrieval curr. user friends based on friendships.
2. Methods were renamed to get cleaner look and add some "naming conventions" into Data Sources and Repositories
3.FriendshipRequestRemoteDataSource getSentSingleAsFlow was tweaked. Parameter userId was removed, because i thought there were no cases when we would need Flow<List<FriendshipRequest>> with status "Sent", for other user except curr. user.
Added getAllAccepted() for getting FriendshipRequests for curr. user with status "Accepted"
4. UsersDataSource.kt added function for getting all Users, similar to getAllAsFlow

scherepiuk commented on app/src/main/java/com/example/geomate/data/repositories/NotificationsRepository.kt:
Leave this file with a single method, to ground different types of notifications

scherepiuk commented on app/src/main/java/com/example/geomate/ui/screens/notifications/NotificationsViewModel.kt:
```diff
@@ -30,4 +31,20 @@ class NotificationsViewModel(
         }
     }
+
+    fun declineRequest(notification: Notification.FriendshipRequest) =
```
Add same methods to `ProfileViewModel`

scherepiuk commented on app/src/main/java/com/example/geomate/ui/screens/notifications/components/Notification.kt:
```diff
@@ -40,19 +47,26 @@ import com.skydoves.landscapist.fresco.FrescoImage
 @Composable
 private fun FriendshipRequest(
     notification: Notification.FriendshipRequest,
+    viewModel: NotificationsViewModel,
```
Pass callbacks instead of whole `ViewModel`

scherepiuk commented on app/src/main/java/com/example/geomate/ui/screens/notifications/components/Notification.kt:
```diff
@@ -116,5 +133,20 @@ private fun FriendshipRequest(
+        if (isAlertDialogVisible) {
+            GeomateAlertDialog(
+                onDismissRequest = {
+
```
white-space

scherepiuk commented on app/src/main/java/com/example/geomate/ui/screens/notifications/components/Notification.kt:
```diff
@@ -116,5 +133,20 @@ private fun FriendshipRequest(
+                    isAlertDialogVisible = false
+                    viewModel.declineRequest(notification)
+                },
+                dialogTitle = "Confirmation Dialog",
```
String resources

scherepiuk reviewed (changes requested):
Minor changes

WretchedBunny committed:
fix: Minor fixes to Notification functionalities
1. Buttons.kt: added AlertDialog to Friendship decline
2. NotificationRepository.kt was returned
4. FriendshipRequest composable retrieves functions instead the whole model
5. Two strings added to string res.
5. Minor visual syntax fixes.
